includes/class-sensei-course-enrolment-provider-results.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * File containing the class Sensei_Course_Enrolment_Provider_Results.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Stores a record for course enrolment checks between a particular user and course.
+ */
+final class Sensei_Course_Enrolment_Provider_Results implements JsonSerializable {
+	/**
+	 * Course enrolment results from providers.
+	 *
+	 * @var array
+	 */
+	private $provider_results = [];
+
+	/**
+	 * Time the results were generated.
+	 *
+	 * @var float
+	 */
+	private $time;
+
+	/**
+	 * Version of enrolment providers. This is a hash of all the enrolment provider versions.
+	 *
+	 * @var string
+	 */
+	private $version;
+
+	/**
+	 * Sensei_Course_Enrolment_Provider_Results constructor.
+	 *
+	 * @param array  $provider_results Course enrolment check results from all providers.
+	 * @param string $version          Version of the enrolment providers that produced this set of results.
```

Comment:
I'll switch out the name/description for now. I originally did have the helper method in this class, but then this class became agnostic about the course provider _class_ names, which made it much more difficult to get the versions.

Reply:
d3420cf883e5923a402f3a5f72b95e5f5f21e7f9

Instead of the second sentence in the description I said where it was generated from. In there it sorts by key.

Reply:
LGTM 👍